### ISSUE TITLE
Failing tests: `pmempool_rm failed: removing file failed`

### ISSUE THREAD

**Issue by draghan** [pmem/pmdk-convert#90]
After merge of https://github.com/pmem/pmdk-convert/pull/88 there are 6 tests failures:
- tx_intr_pool_devdax,
- tx_intr_poolset_local_devdax,
- tx_intr_poolset_local_2devdaxes,
- tx_intr_poolset_local_devdax_rep_devdax,
- tx_intr_poolset_local_2devdaxes_rep_2regfiles,
- tx_intr_poolset_local_2regfiles_rep_2devdaxes.

Reproduction on pmdk-convert revision: dbe788aec74b44d19941fb7d70321cc77e5c648b.
Failures does not occur on revision: d7d9852f2878d31f8de008e6044656a1e18e8447.

Configuration "release" and "debug" are affected.

Tested (and affected) OSes: 
- Red Hat Enterprise Linux 8.2 with 4.18.0-193.1.2.el8_2.x86_64 kernel,
- Fedora 31 (Server Edition) with 5.7.9-100.fc31.x86_64 kernel.

Namespace configuration:
```
ndctl list -X
[
  {
    "dev":"namespace0.2",
    "mode":"devdax",
    "map":"dev",
    "size":1054867456,
    "uuid":"344d27ea-e2c5-449f-b5f1-3091994c235c",
    "daxregion":{
      "id":0,
      "size":1054867456,
      "align":4096,
      "devices":[
        {
          "chardev":"dax0.2",
          "size":1054867456,
          "target_node":2,
          "mode":"devdax"
        }
      ]
    },
    "align":4096
  },                                                                                                                                
  {                                                                                                                                 
    "dev":"namespace0.0",                                                                                                           
    "mode":"devdax",                                                                                                                
    "map":"dev",                                                                                                                    
    "size":1054867456,                                                                                                              
    "uuid":"d52f9357-3f0b-42f0-90e2-db323fadceaf",                                                                                  
    "daxregion":{                                                                                                                   
      "id":0,                                                                                                                       
      "size":1054867456,                                                                                                            
      "align":4096,                                                                                                                 
      "devices":[                                                                                                                   
        {                                                                                                                           
          "chardev":"dax0.0",                                                                                                       
          "size":1054867456,                                                                                                        
          "target_node":2,                                                                                                          
          "mode":"devdax"                                                                                                           
        }                                                                                                                           
      ]                                                                                                                             
    },                                                                                                                              
    "align":4096                                                                                                                    
  }                                                                                                                                 
]
```

**Steps to reproduce:**

1. Build pmdk-convert:

```
$ cd /WORKSPACE

$ git clone https://github.com/pmem/pmdk-convert.git

$ cd pmdk-convert

$ git checkout dbe788aec74b44d19941fb7d70321cc77e5c648b

$ mkdir install

$ mkdir build

$ cd build

$ cmake .. -DCMAKE_INSTALL_PREFIX=/WORKSPACE/install -DDEVELOPER_MODE=1 -DTRACE_TESTS=1 -DCMAKE_BUILD_TYPE=RELEASE -DTESTS_USE_FORCED_PMEM=ON -DDEVICE_DAX_PATHS="/dev/dax0.0 /dev/dax0.2"

$ make -j$(nproc)
```

2. Run pmdk-convert tests:

```
$ ctest -T test --test-output-size-passed 307200 --test-output-size-failed 307200 --no-compress-output --output-on-failure
```

**Expected result:**
The whole tests suite is passing without errors.

**Actual result:**
Tests (tx_intr_pool_devdax, tx_intr_poolset_local_devdax, tx_intr_poolset_local_2devdaxes, tx_intr_poolset_local_devdax_rep_devdax, tx_intr_poolset_local_2devdaxes_rep_2regfiles, tx_intr_poolset_local_2regfiles_rep_2devdaxes) are failing with the following message:
```
...
-- Stdout:

/home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/helpers.cmake(71):  file(READ /home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/build-Debug/tests/tx_intr_pool_devdax/err ERR )
/home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/helpers.cmake(72):  message(STATUS Stderr:
pmempool_rm failed: removing file failed
 )
-- Stderr:
pmempool_rm failed: removing file failed

/home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/helpers.cmake(74):  if(NOT RET EQUAL expectation )
/home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/helpers.cmake(75):  message(FATAL_ERROR /home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/build-Debug/tests/clean_pool /dev/dax0.0 exit code 1 doesn't match expectation 0 )
CMake Error at /home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/helpers.cmake:75 (message):
  
  /home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/build-Debug/tests/clean_pool
  /dev/dax0.0 exit code 1 doesn't match expectation 0
Call Stack (most recent call first):
  /home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/helpers.cmake:80 (execute_arg)
  /home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/tx_intr_pool_devdax.cmake:8 (execute)
  /home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/helpers.cmake:229 (prepare_files)
  /home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/tx_intr_pool_devdax.cmake:30 (test_intr_tx_devdax)
  /home/jenkins-slave/workspace/PMDK-convert-linux/pmdk-convert/tests/tx_intr_pool_devdax.cmake:39 (test_devdax)

```

Log from tests: [tests.log](https://github.com/pmem/pmdk-convert/files/5027748/convert-failed-tests.log)

Whole log with all execution details: [convert-execution.log](https://github.com/pmem/pmdk-convert/files/5027750/convert-execution.log)

**Comment by kasiawasiuta** [comment]
@wlemkows, could you take a look?

**Comment by draghan** [comment]
It is fixed on revision 6b0251c4c967c26f7550559d147ca5b41aaa44c1.
